README.md
# Go forth and build! 🚀

## Tutorial: How to make your first commit 📹:

This video example shows how to:

- Edit your container code
- Raise a new pull request
- Deploy the newest version automatically 🚀

https://user-images.githubusercontent.com/1718624/216366290-9bcac918-8605-4cf5-a98a-75bb8f722966.mp4



By now, you have deployed a container, and in moments, you can visit your app
live!

> Be patient! In ~3 mins your app is live, at your [app url](https://container-e5fwtqs.containers.anotherwebservice.com/). It even has a free SSL/TLS certificate 🔒 you're welcome!

You probably want to add code to your app. Good news, your app is ready right now to start coding, which is simple:

1. Edit your code
2. Commit your code
3. Push your code `git push origin main`

Your app will be automatically re-deployed with the latest code at: https://container-e5fwtqs.containers.anotherwebservice.com/

> You app is deployed already and is working software. Gone are the days of spending weeks coding and then *another* week going to production. No. Go to production *early* and respond to change.

# Getting Started 💻 (locally on your laptop)

> Step 0: You need to download your repo to your computer:

```
git clone git@github.com:divmagic/zoneapi.git
cd container-e5fwtqs
```

> See an error? You might need to setup permissions [here's a guide how to setup repo clone permissions](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)

1. [Install docker](https://docs.docker.com/get-docker/)

2. Start your container locally: `docker-compose up`
3. Visit your app locally: http://127.0.0.1:5000/

## View your app locally

Visit: http://127.0.0.1:5000/

### Rebuild container (locally)
If you make changes to `Dockerfile`, then you need to rebuild your container image. To rebuild the container image:
```
docker-compose build
# or 
docker-compose up --build
```

# Start coding! Which framework did you choose?

Need some help to get started?

- [**Flask** quickstart guide](https://flask.palletsprojects.com/en/2.2.x/quickstart/) ⚗️ 🐍
- [**Django** quide](https://docs.djangoproject.com/en/4.1/topics/http/views/) 📰
- [**Ruby** quickstart guide](https://github.com/KarmaComputing/rails-quickstart) 💎

# Debugging

How do I turn on the debugger?

Enable a breakpoint by adding `breakpoint()` to your code, start your application and run to that point then in a terminal type:

```
docker attach container-e5fwtqs
```
Ta-da! You'll be inside the [Python debugger](https://docs.python.org/3/library/pdb.html#module-pdb) ( ⬅️ Read this!)



## Questions

- How was this built? [All code is here](https://github.com/KarmaComputing/container-hosting)


CHANGELOG.md
# v0.0.3 (Thu Sep 14 2023)

#### ⚠️ Pushed to `main`

- Added deploy.yml file ([@divmagic](https://github.com/divmagic))
- Added release.yml file ([@divmagic](https://github.com/divmagic))
- Added deploy.sh file ([@divmagic](https://github.com/divmagic))
- Added .autorc file ([@divmagic](https://github.com/divmagic))
- Added amber.yaml secrets file ([@divmagic](https://github.com/divmagic))
- Added flask quickstart ([@divmagic](https://github.com/divmagic))

#### Authors: 1

- [@divmagic](https://github.com/divmagic)

---

# v0.0.2 (Thu Sep 14 2023)

#### ⚠️ Pushed to `main`

- Update index.html ([@divmagic](https://github.com/divmagic))

#### Authors: 1

- [@divmagic](https://github.com/divmagic)

---

# v0.0.1 (Thu Sep 14 2023)

:tada: This release contains work from a new contributor! :tada:

Thank you, null[@divmagic](https://github.com/divmagic), for all your work!

#### ⚠️ Pushed to `main`

- Added deploy.yml file ([@divmagic](https://github.com/divmagic))
- Added release.yml file ([@divmagic](https://github.com/divmagic))
- Added deploy.sh file ([@divmagic](https://github.com/divmagic))
- Added .autorc file ([@divmagic](https://github.com/divmagic))
- Added amber.yaml secrets file ([@divmagic](https://github.com/divmagic))
- Added flask quickstart ([@divmagic](https://github.com/divmagic))
- create .docker-compose.yml ([@divmagic](https://github.com/divmagic))
- create README.md ([@divmagic](https://github.com/divmagic))
- Add files via upload ([@divmagic](https://github.com/divmagic))
- Update index.html ([@divmagic](https://github.com/divmagic))
- Update and rename index.html.txt to index.html ([@divmagic](https://github.com/divmagic))

#### Authors: 1

- [@divmagic](https://github.com/divmagic)


scrab.php
<?php
// Generated by curl-to-PHP: http://incarnate.github.io/curl-to-php/
$ch = curl_init();

curl_setopt($ch, CURLOPT_URL, $_GET['link']);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($ch, CURLOPT_POST, 1);
curl_setopt($ch, CURLOPT_ENCODING, 'gzip, deflate');
curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);

$headers = array();
$headers[] = 'Cache-Control: max-age=0';
$headers[] = 'Dnt: 1';
$headers[] = 'Sec-Ch-Ua: ^^Chromium^^\";v=^^\"116^^\",';
$headers[] = 'Sec-Ch-Ua-Mobile: ?1';
$headers[] = 'Sec-Ch-Ua-Platform: ^^Android^^\"\"';
$headers[] = 'Sec-Fetch-Dest: empty';
$headers[] = 'Sec-Fetch-Mode: no-cors';
$headers[] = 'Sec-Fetch-Site: cross-site';
$headers[] = 'Sec-Fetch-User: ?1';
$headers[] = 'Upgrade-Insecure-Requests: 1';
$headers[] = 'User-Agent: Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Mobile Safari/537.36';
$headers[] = 'Referer: https://akw-app.com/';
$headers[] = 'Origin: https://akw-app.com';
$headers[] = 'Connection: keep-alive';
$headers[] = 'Content-Length: 0';
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);

$result = curl_exec($ch);
if (curl_errno($ch)) {
    echo 'Error:' . curl_error($ch);
}


curl_close($ch);


echo html_entity_decode($result);



api.js
const https = require('https');

const options = {
  hostname: 'akw-app.com',
  path: '/',
  method: 'GET',
  headers: {
    'Cache-Control': 'max-age=0',
    'Dnt': '1',
    'Sec-Ch-Ua': '^^Chromium^^";v="116^^",',
    'Sec-Ch-Ua-Mobile': '?1',
    'Sec-Ch-Ua-Platform': '^^Android^^""',
    'Sec-Fetch-Dest': 'empty',
    'Sec-Fetch-Mode': 'no-cors',
    'Sec-Fetch-Site': 'cross-site',
    'Sec-Fetch-User': '?1',
    'Upgrade-Insecure-Requests': '1',
    'User-Agent': 'Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Mobile Safari/537.36',
    'Referer': 'https://akw-app.com/',
    'Origin': 'https://akw-app.com',
    'Connection': 'keep-alive',
    'Content-Length': '0'
  }
};

const req = https.request(options, (res) => {
  let data = '';

  res.on('data', (chunk) => {
    data += chunk;
  });

  res.on('end', () => {
    console.log(htmlDecode(data));
  });
});

req.on('error', (error) => {
  console.error('Error:', error);
});

req.end();

function htmlDecode(input) {
  const doc = new DOMParser().parseFromString(input, 'text/html');
  return doc.documentElement.textContent;
}